### PR TITLE
DYN-4216 - Fixing syntax add GitHub action

### PR DESCRIPTION
**Pull request by carlosgzze**
### Purpose

PR created for fixing spelled word on PR template and adding a new github action for enable notification every time a PR is merged into master. 
Github action will retrieve Release Notes section of the PR and will be send to slack channel. It will be use as a entry for Release Notes wiki page.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Enhancement for enabling a slack notification every time a PR is merged into master.

### Reviewers

@QilongTang @mjkkirschner

### CODE REVIEW

File: .github/workflows/PR_merged.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## This is a basic workflow to help you get started with Actions
```
Comment (mjkkirschner) [inline .github/workflows/PR_merged.yaml:1]: maybe update this comment

File: .github/workflows/PR_merged.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# Controls when the workflow will run
+on:
+   pull_request:
+    branches: [master]
+    types: [closed]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    if: github.event.pull_request.merged == true
```
Comment (mjkkirschner) [inline .github/workflows/PR_merged.yaml:14]: is this also triggered by squash merges?

File: .github/workflows/PR_merged.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          releasenotes=$(cat releasenotes.out)
+          echo "Building JSON message"
+          cat << EOF > ./slack_message.json
+          {
+          "text": "PR $prurl merged into master.\n Content of Release Notes section: $releasenotes"
+          }
+          EOF
+      
+      - name: Notify to Slack
+        uses: opspresso/action-slack@master
```
Comment (mjkkirschner) [inline .github/workflows/PR_merged.yaml:49]: this GitHub action is not widely used and is licensed GPL3 - please find another one with a pre-approved license type (MIT, APACHE 2 etc) - it seems there are other slack web hook actions around.

**Review by mjkkirschner — CHANGES_REQUESTED** [review]

File: .github/workflows/PR_merged.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          echo "Installing requirements"
+          sudo apt update
+          sudo apt install jq -y
+          sudo apt install curl -y
+          bash
+          echo "Getting PR number from merge"
+          git show ${{ github.sha }} > pr_info1.out
+          sed -n -e 's/^.*Merge pull request #//p' pr_info1.out > pr_info2.out
+          sed -n -e 's/from.*$//p' pr_info2.out > pr_info3.out
+          pr_number=$(sed 's/ //g' pr_info3.out)
```
Comment (mjkkirschner) [inline .github/workflows/PR_merged.yaml:32]: did this not work?
https://github.com/actions/checkout/issues/58

Comment (mjkkirschner) [reply .github/workflows/PR_merged.yaml:32]: theres also `github.event.number` apparently.

File: .github/workflows/PR_merged.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          echo "PR number is ${{ github.event.pull_request.number }}"
+          echo "Getting PR URL"
+          curl "https://api.github.com/repos/DynamoDS/Dynamo/pulls/${{ github.event.pull_request.number }}" | jq '._links.html.href' | sed -e 's/"/ /g' | sed 's/ //g' > pr_url_convert.out
+          cat pr_url_convert.out | sed -e 's/\//\\\//g' > pr_url.out
+          prurl=$(cat pr_url.out)
+          echo "Getting Release Notes Section from PR body"
+          curl "https://api.github.com/repos/DynamoDS/Dynamo/pulls/${{ github.event.pull_request.number }}" | jq '.body' | sed -e 's/\\r\\n/ /g' | sed -n -e 's/^.*### Release Notes //p' | sed -n -e 's/### Reviewers.*$//p' > releasenotes.out
+          releasenotes=$(cat releasenotes.out)
+          echo "Building JSON message"
+          cat << EOF > ./slack_message.json
```
Comment (mjkkirschner) [inline .github/workflows/PR_merged.yaml:37]: where is this file?

Comment (carlosgzze) [reply .github/workflows/PR_merged.yaml:37]: The JSON file is created in the ubuntu VM provisioned.

Comment (mjkkirschner) [reply .github/workflows/PR_merged.yaml:37]: Sorry @carlosgzze I'm not understanding, is this .json file created during the run of the github action, or is it created manually beforehand somehow? If beforehand, I think it should be part of the repo or committed somewhere.

Comment (mjkkirschner) [reply .github/workflows/PR_merged.yaml:37]: Did you modify the secret to contain the full URL? Your last instructions to me where to only store the endpoint, so it does not include the domain.

Comment (carlosgzze) [reply .github/workflows/PR_merged.yaml:37]: The JSON file is created during the run of github action and is built using the secrets, PR url and Release Notes section retrieved from PR body.
And it's a good point, I have updated the command to use only the last part of webhook url. Thanks for the heads up.

**Review by mjkkirschner — APPROVED** [review]
thanks for making the changes @carlosgzze LGTM

File: .github/workflows/PR_merged.yaml (hunk 5 above)
Comment (carlosgzze) [reply .github/workflows/PR_merged.yaml:37]: The JSON file is created during the run of the github action and is built using secrets, PR URL and Release Notes section.

**Pull request merged**